assets/src/edit-story/app/media/media3p/typedefs.js

```diff
@@ -0,0 +1,210 @@
+ * action to deselect current
+ * category
+ */
+
+/**
+ * 'media3p/[provider]' state object used by each provider, exposed to the rest
+ * of application through context providers
+ *
+ * @typedef {Object} Media3pProviderContextState
+ * @property {CategoriesState} categories Category state object
+ * @property {boolean} hasMore has more media
+ * @property {boolean} isMediaLoaded is media loaded
+ * @property {boolean} isMediaLoading is media loading
+ * @property {Array.<Media>} media array of media
+ * @property {string} pageToken page token for current page
+ * @property {string} nextPageToken the page token for the next page
+ * @property {number} totalPages total pages
+ */
+
+/**
+ * 'media3p/[provider]' state object used by each provider internally by the
+ * reducer.
+ *
+ * @typedef {Object} Media3pProviderReducerState
+ * @property {CategoriesState} categories Category state object
+ * @property {boolean} hasMore has more media
+ * @property {boolean} isMediaLoaded is media loaded
+ * @property {boolean} isMediaLoading is media loading
+ * @property {Array} media array of media
+ * @property {string} pageToken page token for current page
+ * @property {string} nextPageToken the page token for the next page
+ * @property {number} totalPages total pages
+ */
+
+/**
+ *
+ * @typedef {Object} Media3pProviderContext
+ * @property {Media3pProviderContextState} state state of a single provider
+ * @property {Media3pProviderContextActions} actions actions for the provider
+ */
+
+/**
+ * media3p/ typedefs ===========================================================
+ */
+
+/**
+ * 'media3p' root action typedefs
+ *
+ * @typedef {(payload: {searchTerm: string}) => undefined} SetSearchTermFn
+ * @typedef {(payload: {provider: string}) => undefined} SetSelectedProviderFn
+ */
+
+/**
+ * 'media3p' root actions in the Media3pContext object.
+ *
+ * @typedef {Object} Media3pContextActions
+ * @property {SetSearchTermFn} setSearchTerm Sets current search term
+ * @property {SetSelectedProviderFn} setSelectedProvider Sets provider
+ */
+
+/**
+ * 'media3p' root state in the Media3pContext object.
+ *
+ * @typedef {Object} Media3pContextState
+ * @property {string} searchTerm search term for media3p
+ * @property {string} selectedProvider current selected provider
+ * @property {Media3pProviderContextState} unsplash state for unsplash
+ * @property {Media3pProviderContextState} coverr state for coverr
+ */
+
+/**
+ * Context object representing all of media3p's state and actions.
+ *
+ * @typedef {Object} Media3pContext
+ * @property {Media3pContextState} state Non-provider-specific state
+ * @property {Media3pContextActions} actions Non-provider-specific actions
+ * @property {Media3pProviderContext} unsplash Unsplash state and actions
+ * @property {Media3pProviderContext} coverr Coverr state and actions
+ */
+
+/**
+ * The 'media3p' fragment of the state returned from
+ * `useMediaReducer`
```

Comment:
[nit]: unnecessary line wrap and period at end of sentence (here and below).

Reply:
Done.